Kernel/Devices/Storage/Management.cpp

```diff
@@ -207,18 +207,26 @@ UNMAP_AFTER_INIT ErrorOr<NonnullOwnPtr<Partition::PartitionTable>> StorageManage
 
 UNMAP_AFTER_INIT void StorageManagement::enumerate_disk_partitions()
 {
-    VERIFY(!m_storage_devices.is_empty());
-    for (auto& device : m_storage_devices) {
-        auto partition_table_or_error = try_to_initialize_partition_table(device);
+    // FIXME: Find a way to not allocate memory in this method!
```

Comment:
Why is it important not to allocate here? And `::enumerate_disk_partitions` is only called during initialization, we don't even strictly need locking here afaic.